package/rmt-server.spec

```diff
@@ -180,6 +180,9 @@ rm -f %{buildroot}%{lib_dir}/vendor/bundle/ruby/*/extensions/*/*/*/gem_make.out
 rm -f %{buildroot}%{lib_dir}/vendor/bundle/ruby/*/extensions/*/*/*/mkmf.log
 find %{buildroot}%{lib_dir}/vendor/bundle/ruby/*/gems/yard*/ -type f -exec chmod 644 -- {} +
 
+# give _rmt user access to /run/lock
+usermod -a -G lock _rmt
```

Comment:
when building:

```
[   22s] + usermod -a -G lock _rmt
[   22s] /var/tmp/rpm-tmp.Ugw9P7: line 104: usermod: command not found
```

I'm still trying to find the proper way to do this...

Reply:
Ah, ok, I think we should add it here, when creating the user

https://github.com/SUSE/rmt/blob/936a0a37e0756a3ebec962a6e0c241428f932b75/package/rmt-server.spec#L219

Reply:
Something like:

```
%{_sbindir}/useradd -g %{rmt_group} -G lock -s /bin/false -r \
```

I'm testing this right now.

Reply:
Not working for me =/

It didn't add the user to the `lock` group.

And then I even added the user manually (with `usermod`), and still no cigar. Strange!

![image](https://user-images.githubusercontent.com/2944/40066946-7fcbffe4-5865-11e8-93e6-e2404bbe4cc0.png)

Reply:
Give it a try with `%files`?

Reply:
The problem is not logging in and out, but that `usermod` is not usable within the RPM spec =/
And it's not possible to "log in" as `_rmt`, that's just the user that the `rmt-cli` process runs as. So even if you're root, `rmt-cli` will run as `_rmt`...

Reply:
okay, so actually I think it doesn't make sense to create `/var/run/lock/rmt` or `/run/lock/rmt` or `/var/lock/rmt` in the RPM.

```
> ll /var/
lrwxrwxrwx  1 root root    9 Apr 13 15:28 lock -> /run/lock/
lrwxrwxrwx  1 root root    4 Apr 13 15:28 run -> /run/
```

and 

```
> mount | grep tmpfs
tmpfs on /run type tmpfs (rw,nosuid,nodev,mode=755)
```

Basically, `/run` is a `tmpfs` so anything that gets created there will be deleted upon system restart. It makes no sense for the RPM to create something there.

So I think that adding `_rmt` to the `lock` group and just writing to `/var/lock` should work. I just don't know how to make that work :cry: 
This attempt doesn't work: https://github.com/SUSE/rmt/commit/1057783e54d0c8a7d62b9e83b585a85d75bb5fa6

`_rmt` does not get added to the `lock` group.
And even adding it manually with `usermod`, it still complains about permission denied.

Strangely, adding my own user to `lock` did let me write to `/var/lock`.

Reply:
@lagartoflojo I thing the issue might be the way we switch the user here:

https://github.com/SUSE/rmt/blob/17c2366e9a48af93d0f1e01118c8a9e9364f9398/bin/rmt-cli#L15-L20

That sets the group only to `nginx`, the process will not have `lock` group. I suggest you investigate `supplementary groups` and also this: https://ruby-doc.org/core-2.1.1/Process.html#method-c-groups

Reply:
I see... Okay, @Sergeykot you can investigate that, this is your PR after all hehe..
But I'm thinking that it would be easier to just write the lock file to `Rails.root + '/tmp'` as it was before.

Reply:
`Rails.root + '/tmp'` might be problematic -- I think some users might be mounting `/usr` in read-only mode, and this is why the whole convention exists about putting stuff that the application generates into `/var`. `/tmp` should work, though.